File: app/src/main/java/com/sd/www/cache/converter/GsonObjectConverter.java
package com.sd.www.cache.converter;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.sd.lib.cache.Cache;

/**
 * Created by Administrator on 2017/8/29.
 */
public class GsonObjectConverter implements Cache.ObjectConverter {
    private final Gson gson = new Gson();

    @NonNull
    @Override
    public byte[] objectToByte(@NonNull Object value) throws Exception {
        return gson.toJson(value).getBytes();
    }

    @NonNull
    @Override
    public <T> T byteToObject(@NonNull byte[] bytes, @NonNull Class<T> clazz) throws Exception {
        return gson.fromJson(new String(bytes), clazz);
    }
}

File: lib/src/main/java/com/sd/lib/cache/simple/GsonObjectConverter.kt
package com.sd.lib.cache.simple

import com.google.gson.Gson
import com.sd.lib.cache.Cache.ObjectConverter

internal class GsonObjectConverter : ObjectConverter {
    private val _gson = Gson()

    override fun objectToByte(value: Any): ByteArray {
        return _gson.toJson(value).toByteArray()
    }

    override fun <T> byteToObject(bytes: ByteArray, clazz: Class<T>): T {
        return _gson.fromJson(String(bytes), clazz)
    }
}